spec/core/ics-002-client-semantics/README.md

```diff
@@ -67,6 +68,22 @@ appropriate gas metering mechanism to charge for compute and storage. On a host
 which supports WASM execution, for example, the validity predicate and misbehaviour predicate
 could be provided as executable WASM functions when the client instance is created.
 
+Conditional clients can express dependencies on existing client instances. The idea is here is that modular blockchains change
+the way a logical blockchain can be expressed. Since a logical blockchain can be broken up into its constituent parts, 
+a single client can become multiple clients. This results in greater state layer diversity for IBC.
+
+For example, we could have a different client for each of the following functional layers of the blockchain stack:
+* Data availability
+* Sequencing or transaction ordering
+* Execution
```

Comment:
For IBC, we only really care about execution. ie the `app state` of the blockchain.

In the modular world, in order to have a client following execution, do we need a client of the other layers?

Reply:
The issue from my POV is that to prove execution - you also need to prove sequencing (TO) & data availability (DA) etc. 

For example, in the optimistic rollup (ORU) on Celestia case in the simple case you could have:
- A consensus proof from a Celestia client (TMLC) for proof of DA and TO
- A fraud proof from an ORU client (optimistic-LC) for proof of execution
  - ORU client depends on Celestia client
  - Accepts headers that and starts a fraud window only when the header block has been persisted to the Celestia network

Also, with new shared security primitives like babylon chain (checkpointing on BTC) - you can have dependencies on a babylon client for proof of security/checkpoint.

Reply:
in this specific case that you're talking about for ORUs on Celestia, by proving execution are you not also proving DA and TO? since if the data was not available and the txs were not ordered properly (which is verified on the client side), the rollup nodes would reject those txs anyway?

nevertheless, i think proving DA would be necessary since each Celestia rollup can in theory have their own fork choice rule (i.e. 'Sovereign rollups')

Reply:
If you don't prove DA before initiating the fraud window you can get the following:
- Accept header optimistically w/o ensuring data is available
- Start fraud window
- Data is withheld so no fraud prover can actually generate a fraud proof
- Invalid header gets acceptable after fraud window period